### ISSUE TITLE
x509 package fails to build on darwin with Go 1.10 beta2

### ISSUE THREAD

**Issue by ericchiang** [google/certificate-transparency-go#131]
Seeing these errors on tip (52c0590bd3b4b80c5497005b0f47e10557425eeb)

```
../../../google/certificate-transparency-go/x509/root_cgo_darwin.go:210: cannot use nil as type _Ctype_CFDataRef in assignment
../../../google/certificate-transparency-go/x509/root_cgo_darwin.go:211: cannot use nil as type _Ctype_CFDataRef in assignment
../../../google/certificate-transparency-go/x509/root_cgo_darwin.go:221: cannot convert nil to type _Ctype_CFDataRef
```

**Comment by daviddrysdale** [comment]
Closing for now, we'll investigate if this is still a problem after Go 1.10 is released.

**Comment by daviddrysdale** [comment]
Upstream commit [b868616b63a8](https://github.com/golang/go/commit/b868616b63a82a4f5917400b2df63a19ebe041e2) should fix when we merge in upstream Go 1.10 changes.

**Comment by daviddrysdale** [comment]
Looks like Go 1.10 is now out, but we can't just blindly pull in the relevant commit -- discussion over at #153 and #154
